bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3872,6 +3872,11 @@ ReplicaImp::ReplicaImp(const LoadedReplicaData &ld,
   metric_last_agreed_view_.Get().Set(lastAgreedView);
   metric_current_primary_.Get().Set(getCurrentView() % config_.getnumReplicas());
 
+  if (config_.timeServiceEnabled) {
```

Comment:
There are 3 constructors for ReplicaImp:
1) ReplicaImp::ReplicaImp(bool firstTime,
2) ReplicaImp::ReplicaImp(const ReplicaConfig &config,
3) ReplicaImp::ReplicaImp(const LoadedReplicaData &ld,
I think this is needed for both firstTime (true or false).
Also this requires ReplicaConfig. So its not possible to add it in ctor (1). So it need to be added to both ctors (1) and (2). My suggestion is to create a separate function which will take the time service enabled parameter and aggregator and then that function should be called from both ctor (2) and (3)

Reply:
great catch thanks